localstack/services/cloudwatch/alarm_scheduler.py

```diff
@@ -115,10 +118,13 @@ def _is_alarm_supported(self, alarm_details: MetricAlarm) -> bool:
         return True
 
 
-def get_metric_alarm_details_for_alarm_arn(alarm_arn: str) -> MetricAlarm:
+def get_metric_alarm_details_for_alarm_arn(alarm_arn: str) -> Optional[MetricAlarm]:
     alarm_name = arns.extract_resource_from_arn(alarm_arn).split(":", 1)[1]
     client = get_cloudwatch_client_for_region_of_alarm(alarm_arn)
-    return client.describe_alarms(AlarmNames=[alarm_name])["MetricAlarms"][0]
+    response = client.describe_alarms(AlarmNames=[alarm_name])["MetricAlarms"]
```

Comment:
nit: would probably rename `response` to something like `metric_alarms`. Being this one a list, lines 125 to 127 could be shortened to `return metric_alarms[0] if metrics_alarms else None`.

Reply:
yes, you are right - that would be much cleaner :) thanks!